src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts

```diff
@@ -63,43 +70,52 @@ export class TaskAssignmentContainerComponent implements OnInit {
     defaultSortFieldName: 'dueDate',
     fields: this.fields,
   };
-  public ngOnInit(): void {
-    // Set up the default sorting.
-    this.sortedBy = {
-      fieldName: this.taskServiceConfig.defaultSortFieldName,
-      order: this.taskServiceConfig.defaultSortDirection
-    };
 
+  public ngOnInit(): void {
+    this.initForm();
     // Get the task from the route, which will have been put there by the resolver.
     const task = this.route.snapshot.data.taskAndCaseworkers.task.task;
     const caseworkers = this.route.snapshot.data.taskAndCaseworkers.caseworkers;
     task.assigneeName = getAssigneeName(caseworkers, task.assignee);
-    this.tasks = [ task ];
+    this.tasks = [task];
     this.verb = this.route.snapshot.data.verb as TaskActionType;
     this.successMessage = this.route.snapshot.data.successMessage as InfoMessage;
     if (task.assignee) {
       const names: string[] = task.assignee.split(' ');
       const firstName = names.shift();
       const lastName = names.join(' ');
-      this.excludedCaseworkers = [ { firstName, lastName } as Caseworker ];
+      this.excludedCaseworkers = [{ firstName, lastName } as Caseworker];
     }
     if (task.location) {
       this.location = { locationName: task.location } as Location;
     }
   }
 
+  public ngOnDestroy(): void {
+    this.assignTask$.unsubscribe();
+  }
+
+  public initForm(): void {
+    this.formGroup = new FormGroup({
+      caseWorkerName: new FormControl()
+    });
+  }
+
   public assign(): void {
     if (!this.caseworker) {
       this.error = NAME_ERROR;
       return;
     }
     this.error = null;
-    this.taskService.assignTask(this.tasks[0].id, {userId: this.caseworker.idamId}).subscribe(() => {
-      this.reportSuccessAndReturn();
-    }, error => {
-      const handledStatus = handleFatalErrors(error.status, this.router);
-      if (handledStatus > 0) {
-        this.reportUnavailableErrorAndReturn();
+
+    this.assignTask$ = this.taskService.assignTask(this.tasks[0].id, { userId: this.caseworker.idamId }).subscribe({
+      next: () => this.reportSuccessAndReturn(),
+      error: (error: any) => {
+        const handledStatus = handleFatalErrors(error.status, this.router);
+
+        if (handledStatus > 0) {
+          this.reportUnavailableErrorAndReturn();
+        }
```

Comment:
Can this subscribe block move to ngOnInit() lifecycle? Otherwise If there is strong reason it's still subcribed within a button click event.

Reply:
But it needs data that is only available when you click assign, which is not available from ngOnInit

Reply:
I will approve temporarily and will revisit this later